pkgs/dart_services/lib/src/flutter_genui.dart

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2014, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:http/http.dart' as http;
+import 'package:logging/logging.dart';
+
+final _logger = Logger('genui');
+
+class GenUi {
+  static const _apiKeyVarName = 'GENUI_API_KEY';
+  late final String _genuiApiKey;
+
+  GenUi() {
+    _genuiApiKey = Platform.environment[_apiKeyVarName] ?? '';
+    if (_genuiApiKey.isEmpty) {
+      _logger.warning('$_apiKeyVarName not set; genui features DISABLED');
+    } else {
+      _logger.info('$_apiKeyVarName set; genui features ENABLED');
+    }
+  }
+
+  Future<String> generateCode({required String prompt}) async {
+    final response = await _requestGenUI(prompt: prompt);
+
+    if (response.statusCode != 200) {
+      throw Exception('Failed to generate ui: ${response.body}');
+    }
+
+    final decoded = jsonDecode(response.body) as Map<String, dynamic>;
+    final flutterCode = decoded['flutterCode'] as String;
+
+    return flutterCode;
+  }
+
+  Future<http.Response> _requestGenUI({required String prompt}) async {
+    if (_genuiApiKey.isEmpty) {
+      throw Exception('Missing environment variable: $_apiKeyVarName');
+    }
+
+    return http.post(
+      Uri.parse(
+        'https://autopush-devgenui.sandbox.googleapis.com/v1beta1/firstparty/generateidecode?key=$_genuiApiKey',
```

Comment:
put this in a const

Reply:
Why? It makes sense to define consts for values, that are used in at least two places. 
Otherwise it increases code size and reduce readability.